client/domain/CorrectnessTestObjectFactory.js

```diff
@@ -31,6 +31,17 @@ tie.factory('CorrectnessTestObjectFactory', [
     };
 
     CorrectnessTest.prototype.matchesOutput = function(output) {
+    // If allowOutputs should be arrays and learner's code returns an array
```

Comment:
Modified as your suggested. It's cleaner and more efficient. No traps here, just I didn't know JS/angular before and wasn't ware that angular had this beauty...